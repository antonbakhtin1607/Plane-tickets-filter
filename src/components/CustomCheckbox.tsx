import React from 'react';
import { Checkbox, FormControlLabel } from '@mui/material';
import CheckBoxOutlineBlankOutlinedIcon from '@mui/icons-material/CheckBoxOutlineBlankOutlined';
import CheckBoxOutlinedIcon from '@mui/icons-material/CheckBoxOutlined';

import { styledTheme } from '../theme';

type CustomCheckboxProps = {
  checked: boolean;
  label: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
};

const CustomCheckbox: React.FC<CustomCheckboxProps> = ({
  checked,
  label,
  onChange,
}) => (
  <FormControlLabel
    control={
      <Checkbox
        checked={checked}
        icon={<CheckBoxOutlineBlankOutlinedIcon />}
        checkedIcon={<CheckBoxOutlinedIcon />}
        onChange={onChange}
        sx={{
          color: styledTheme.colors.blue,
          '&.Mui-checked': {
            color: styledTheme.colors.blue,
          },
        }}
      />
    }
    label={label}
  />
);

export default CustomCheckbox;
