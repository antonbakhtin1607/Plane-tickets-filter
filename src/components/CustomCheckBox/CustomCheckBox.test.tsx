import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import CustomCheckbox from './CustomCheckbox';

describe('CustomCheckbox Component', () => {
  const mockOnChange = jest.fn();

  it('renders the checkbox with the correct label', () => {
    render(<CustomCheckbox checked={false} label="Test Label" onChange={mockOnChange} />);
    expect(screen.getByLabelText('Test Label')).toBeInTheDocument();
  });

  it('applies the checked state correctly', () => {
    render(<CustomCheckbox checked={true} label="Checked" onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    expect(checkbox).toBeChecked();
  });

  it('calls the onChange handler when clicked', () => {
    render(<CustomCheckbox checked={false} label="Click Test" onChange={mockOnChange} />);
    const checkbox = screen.getByRole('checkbox');
    fireEvent.click(checkbox);
    expect(mockOnChange).toHaveBeenCalledTimes(1);
  });
});
