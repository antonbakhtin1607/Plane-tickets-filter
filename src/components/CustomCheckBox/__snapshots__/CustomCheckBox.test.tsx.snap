// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCheckbox Component applies the checked state correctly 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium Mui-checked MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1frrnlk-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        checked=""
        class="PrivateSwitchBase-input css-j8yymo"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckBoxOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2m0 16H5V5h14zM17.99 9l-1.41-1.42-6.59 6.59-2.58-2.57-1.42 1.41 4 3.99z"
        />
      </svg>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      Checked
    </span>
  </label>
</DocumentFragment>
`;

exports[`CustomCheckbox Component calls the onChange handler when clicked 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1frrnlk-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
        />
      </svg>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      Click Test
    </span>
  </label>
</DocumentFragment>
`;

exports[`CustomCheckbox Component calls the onChange handler when clicked 2`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1frrnlk-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
        />
      </svg>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      Click Test
    </span>
  </label>
</DocumentFragment>
`;

exports[`CustomCheckbox Component renders the checkbox with the correct label 1`] = `
<DocumentFragment>
  <label
    class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-19opgx6-MuiFormControlLabel-root"
  >
    <span
      class="MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium PrivateSwitchBase-root MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium MuiCheckbox-root MuiCheckbox-colorPrimary MuiCheckbox-sizeMedium css-1frrnlk-MuiButtonBase-root-MuiCheckbox-root"
    >
      <input
        class="PrivateSwitchBase-input css-j8yymo"
        data-indeterminate="false"
        type="checkbox"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
        data-testid="CheckBoxOutlineBlankOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 5v14H5V5zm0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2"
        />
      </svg>
    </span>
    <span
      class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-rizt0-MuiTypography-root"
    >
      Test Label
    </span>
  </label>
</DocumentFragment>
`;
